mujoco_warp/_src/collision_gjk.py

```diff
@@ -89,37 +89,25 @@ class SupportPoint:
 
 
 @wp.func
-def _discrete_geoms(g1: int, g2: int) -> bool:
+def _discrete_geoms(geom1: Geom, geom2: Geom, g1: int, g2: int) -> bool:
+  if geom1.margin > 0.0 or geom2.margin > 0.0:
+    return False
   return (g1 == int(GeomType.MESH.value) or g1 == int(GeomType.BOX.value) or g1 == int(GeomType.HFIELD.value)) and (
     g2 == int(GeomType.MESH.value) or g2 == int(GeomType.BOX.value) or g2 == int(GeomType.HFIELD.value)
   )
 
 
 @wp.func
-def _support_margin(geom: Geom, geomtype: int, dir: wp.vec3) -> SupportPoint:
+def _support(geom: Geom, geomtype: int, dir: wp.vec3) -> SupportPoint:
   sp = SupportPoint()
   sp.cached_index = -1
   sp.vertex_index = -1
   if geomtype == int(GeomType.SPHERE.value):
-    sp.point = geom.pos
+    sp.point = geom.pos + (0.5 * geom.margin) * geom.size[0] * dir
```

Comment:
it might be helpful to add a comment noting that `geom.margin` and `geom.size` have been modified in `gjk`?

(for an unmodified sphere `geom` i would have expected the margin to be added to the radius (size[0]))